CryptoPkg/Library/BaseCryptLib/Pk/CryptX509.c

```diff
@@ -1071,6 +1071,7 @@ X509GetSerialNumber (
   }
 
   if (SerialNumber != NULL) {
+    *SerialNumberSize = MIN ((UINTN)Asn1Integer->length, *SerialNumberSize);
     CopyMem (SerialNumber, Asn1Integer->data, *SerialNumberSize);

```

Comment:
```suggestion
    CopyMem (SerialNumber, Asn1Integer->data, (UINTN)Asn1Integer->length);
```
After offline discussion, the change itself is not an issue given the check performed a few lines above. But it is suggested to change the update to above shape instead of updating SerialNumberSize twice for better clarity.

Reply:
Updated.